x/delayedack/keeper/invariants_test.go

```diff
@@ -0,0 +1,243 @@
+	// test fraud
+	for rollapp := range seqPerRollapp {
+		err := suite.App.DelayedAckKeeper.HandleFraud(suite.Ctx, rollapp)
+		suite.Require().Nil(err)
+		break
+	}
+
+	// check invariant
+	msg, bool := keeper.AllInvariants(suite.App.DelayedAckKeeper)(suite.Ctx)
+	suite.Require().False(bool, msg)
+}
+
+func (suite *DelayedAckTestSuite) TestRollappPacketsCasesInvariant() {
+	suite.SetupTest()
+	ctx := suite.Ctx
+	rollapp := "rollapp1"
+
+	cases := []struct {
+		name             string
+		frozenRollapp    bool
+		allFinalized     bool
+		packet           commontypes.RollappPacket
+		packet2          commontypes.RollappPacket
+		expectedIsBroken bool
+	}{
+		{
+			"successful invariant check",
+			false,
+			false,
+			commontypes.RollappPacket{
+				RollappId:   rollapp,
+				Status:      commontypes.Status_FINALIZED,
+				ProofHeight: 5,
+				Packet:      getNewTestPacket(1),
+			},
+			commontypes.RollappPacket{
+				RollappId:   rollapp,
+				Status:      commontypes.Status_PENDING,
+				ProofHeight: 15,
+				Packet:      getNewTestPacket(2),
+			},
+			false,
+		},
+		{
+			"successful revert check",
+			true,
+			false,
+			commontypes.RollappPacket{
+				RollappId:   rollapp,
+				Status:      commontypes.Status_FINALIZED,
+				ProofHeight: 5,
+				Packet:      getNewTestPacket(1),
+			},
+			commontypes.RollappPacket{
+				RollappId:   rollapp,
+				Status:      commontypes.Status_REVERTED,
+				ProofHeight: 15,
+				Packet:      getNewTestPacket(2),
+			},
+			false,
+		},
+		{
+			"error non-finalized packet",
+			false,
+			true,
+			commontypes.RollappPacket{
+				RollappId:   rollapp,
+				Status:      commontypes.Status_FINALIZED,
+				ProofHeight: 5,
+				Packet:      getNewTestPacket(1),
+			},
+			commontypes.RollappPacket{
+				RollappId:   rollapp,
+				Status:      commontypes.Status_PENDING,
+				ProofHeight: 15,
+				Packet:      getNewTestPacket(2),
+			},
+			true,
+		},
+		{
+			"wrong invariant revert check",
+			true,
+			false,
+			commontypes.RollappPacket{
+				RollappId:   rollapp,
+				Status:      commontypes.Status_FINALIZED,
+				ProofHeight: 5,
+				Packet:      getNewTestPacket(1),
+			},
+			commontypes.RollappPacket{
+				RollappId:   rollapp,
+				Status:      commontypes.Status_PENDING,
+				ProofHeight: 15,
+				Packet:      getNewTestPacket(2),
+			},
+			true,
+		},
+	}
+	for _, tc := range cases {
+		suite.Run(tc.name, func() {
+			// create rollapp
+			suite.CreateRollappWithName(rollapp)
+			proposer := suite.CreateDefaultSequencer(ctx, rollapp)
+
+			// update state infos
+			stateInfo := rollapptypes.StateInfo{
+				StateInfoIndex: rollapptypes.StateInfoIndex{
+					RollappId: rollapp,
+					Index:     1,
+				},
+				StartHeight: 1,
+				NumBlocks:   10,
+				Status:      commontypes.Status_FINALIZED,
+				Sequencer:   proposer,
+			}
+			stateInfo2 := rollapptypes.StateInfo{
+				StateInfoIndex: rollapptypes.StateInfoIndex{
+					RollappId: rollapp,
+					Index:     2,
+				},
+				StartHeight: 10,
```

Comment:
should this definitely be 10 and not 11?

Reply:
true, fixed.